src/lib/filehandler.rs

```diff
@@ -1,16 +1,52 @@
-use std::fs::File;
+use std::fs::*;
+use std::io::*;
+use basic_types::instruction_set::*;
+use basic_types::instruction::*;
+#[derive(Debug)]
 pub struct FileHandler {
     path: String,
-    file: File,
+    buf: BufReader<File>,
 }
 impl FileHandler {
     pub fn new(path: String) -> FileHandler {
         let file = File::open(&path).unwrap();
+        let f = BufReader::new(file);
         return FileHandler {
             path: path,
-            file: file,
+            buf: f,
         };
     }
+    pub fn read_instruction(&mut self) -> Option<Result<String>> {
```

Comment:
nope, not necessarily check [this](https://github.com/oddcoder/sickassembler/blob/master/src/lib/pass_two/translator.rs#L37) out, ( in case the error can be statically written in code )
and [this](https://github.com/oddcoder/sickassembler/blob/master/src/lib/basic_types/instruction.rs#L162) (in case the error string isn't determined at compile time)